src/java.base/share/classes/java/util/ImmutableCollections.java

```diff
@@ -1844,6 +1844,17 @@ public int size() {
                 public Iterator<Map.Entry<K, StableValue<V>>> iterator() {
                     return new StableEnumMapIterator();
                 }
+
+                @Override
+                public int hashCode() {
+                    return System.identityHashCode(this);
+                }
+
+                @Override
+                public boolean equals(Object o) {
+                    return o == this;
```

Comment:
These implementations are violations to the Set contracts; Set's hash code should be its elements' sum (thus an entry set's hash code is equivalent to its map's hash) and equals should check if all elements are present. This also makes two entry sets from two `entrySet()` calls not equal (at least before valhalla)

Reply:
Good catch. Thank you for finding this!